src/util/sss_semanage.c

```diff
@@ -73,6 +74,44 @@ static void sss_semanage_close(semanage_handle_t *handle)
     semanage_handle_destroy(handle);
 }
 
+static int sss_is_selinux_managed(semanage_handle_t *ext_handle)
+{
+    int ret;
+    semanage_handle_t *handle;
+
+    if (ext_handle == NULL) {
+        handle = semanage_handle_create();
+        if (handle == NULL) {
+            DEBUG(SSSDBG_CRIT_FAILURE, "Cannot create SELinux management handle\n");
+            return EIO;
+        }
+
+        semanage_msg_set_callback(handle,
+                                  sss_semanage_error_callback,
+                                  NULL);
+    } else {
+        handle = ext_handle;
+    }
```

Comment:
May I know when ext_handle can/cannot be NULL

IMHO handle should be always initialized

```
115 static int sss_semanage_init(semanage_handle_t **_handle)
116 {
117     int ret;
118     semanage_handle_t *handle = NULL;
119 
120     handle = semanage_handle_create();
121     if (!handle) {
122         DEBUG(SSSDBG_CRIT_FAILURE, "Cannot create SELinux management handle\n");
123         ret = EIO;
124         goto done;
125     }
126 
127     semanage_msg_set_callback(handle,
128                               sss_semanage_error_callback,
129                               NULL);
130 
131     ret = sss_is_selinux_managed(handle);
```

Reply:
You can pass NULL to the sss_is_selinux_managed, In which case a temporary handle will be created or you can pass existing handle in which case no new handle will be created.

The NULL is used in get_seuser and the existing handle is used sss_semanage_init. I think it is worth adding a comment to the code to clarify this.

Reply:
I think it worth to add doc comment that you must not pass NULL as a handler and keep function simpler.
Function still can return EINVAL if handle is NULL

Reply:
Ok, this makes sense. The handler can be created in the caller all the time.